msg/vehicle_status_flags.msg

```diff
@@ -9,50 +8,50 @@ uint64 timestamp				# time since system start (microseconds)
 # Per-mode requirements
 uint32 mode_req_angular_velocity
 uint32 mode_req_attitude
+uint32 mode_req_local_alt
 uint32 mode_req_local_position
 uint32 mode_req_local_position_relaxed
 uint32 mode_req_global_position
-uint32 mode_req_local_alt
 uint32 mode_req_mission
 uint32 mode_req_offboard_signal
 uint32 mode_req_home_position
 uint32 mode_req_prevent_arming    # if set, cannot arm while in this mode
 uint32 mode_req_other             # other requirements, not covered above (for external modes)
 
 
-bool calibration_enabled
-bool pre_flight_checks_pass		# true if all checks necessary to arm pass
-bool auto_mission_available		# Mission available
-bool angular_velocity_valid 		# Angular velocity valid
-bool attitude_valid 			# Attitude valid
-bool local_altitude_valid               # Local altitude valid
-bool local_position_valid		# Local position estimate valid
-bool local_position_valid_relaxed	# Local position with reduced accuracy requirements (e.g. flying with optical flow)
-bool local_velocity_valid		# set to true by the commander app if the quality of the local horizontal velocity data is good enough to use for navigation
-bool global_position_valid		# Global position estimate valid
-bool gps_position_valid
-bool home_position_valid		# Home position valid
-
-bool offboard_control_signal_lost # Offboard signal lost
-bool geofence_violated 				# Geofence violated
-
-bool rc_signal_lost # RC signal lost
-bool data_link_lost # Data link lost
-bool mission_failure # Mission failure
-bool rc_calibration_in_progress
-bool vtol_transition_failure                        # VTOL transition failed
-
-bool battery_low_remaining_time
-bool battery_unhealthy
-
-bool wind_limit_exceeded
-bool flight_time_limit_exceeded
-
-# failure detector
-bool fd_esc_arming_failure
-bool fd_critical_failure # Critical failure (attitude/altitude limit exceeded, or external ATS)
-bool fd_imbalanced_prop # Imbalanced propeller detected
-bool fd_motor_failure
-
-uint8 battery_warning # Battery warning level
+# Mode requirements
+bool angular_velocity_invalid         # Angular velocity invalid
```

Comment:
This is kind of pedantic, but why invert these flags? I think for the majority of the usage it's slightly more intuitive as we care if it's valid (rather than not invalid) and the default initialised state makes sense (everything false).

Reply:
There's both angles to this. I wanted to have consistency within the flags, so that `true -> might trigger failsafe`. This allows consistent handling within the failsafe state machine, the web simulator and automated log analysis tooling.

Reply:
Given the way people extend this stuff in private I would feel more comfortable if the default value was inert and you didn't need to explicitly invalidate.

Once we get a bit closer to ROS2 I'm hoping we can introduce default values in our .msgs.